optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -131,6 +136,29 @@ def _from_pretrained(
         model_save_dir = Path(model_cache_path).parent
         inc_config = None
         msg = None
+        try:
+            quantization_config = WeightOnlyQuantConfig.from_pretrained(model_id)
+            if getattr(
+                quantization_config, "algorithm", None
+            ) is not None and quantization_config.algorithm.lower() in ["rtn", "gptq", "awq", "autoaround"]:
+                if not is_intel_extension_for_transformers_available():
+                    raise ImportError(
+                        "Didn't find out intel-etension-for-transformers package. "
+                        "Please install packages: pip install intel-etension-for-transformers and pip install peft."
+                    )
+                return ITREX_WOQ_MODEL.from_pretrained(
+                    pretrained_model_name_or_path=model_id,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    force_download=force_download,
+                    cache_dir=cache_dir,
+                    local_files_only=local_files_only,
+                    subfolder=subfolder,
+                    trust_remote_code=trust_remote_code,
+                    **kwargs,
+                )
```

Comment:
I think it's a great idea to have the integration in `INCModel`, but would prefer to only do it in case a `quantization_config` is provided when loading the model with `from_pretrained` https://huggingface.co/docs/transformers/v4.39.1/en/quantization#quantization such as :


```python
quantized_model = INCModelForCausalLM.from_pretrained(... quantization_config=quantization_config)
```

Reply:
Based on my understanding, the function :quantized_model = INCModelForCausalLM.from_pretrained(... quantization_config=quantization_config) will quantize a fp32( or fp16) model to int4 model, if we want get a quantized model in model hub which be uploaded before, it should be no quantization_config. So I will temporarily use the above method, and wait for the next version of ITREX, where quantization_config is included in the config, so that we can distinguish whether it is a quantified model or not with config.